src/Storages/ObjectStorageQueue/ObjectStorageQueueOrderedFileMetadata.cpp

```diff
@@ -218,6 +216,131 @@ bool ObjectStorageQueueOrderedFileMetadata::getMaxProcessedFile(
     return false;
 }
 
+std::optional<ObjectStorageQueueOrderedFileMetadata::LastProcessedInfo> ObjectStorageQueueOrderedFileMetadata::getLastProcessedFile(
+    Coordination::Stat * stat,
+    bool check_failed,
+    LoggerPtr log_)
+{
+    std::optional<std::string> processed_node_hive_partitioning_path;
+    if (is_path_with_hive_partitioning)
+        processed_node_hive_partitioning_path = std::filesystem::path(processed_node_path) / getHivePart(path);
```

Comment:
Please add a small comment here in code about what we store in `processed_path` vs `processed_hive_partitioning_path`